### ISSUE TITLE
Chat history receive fail and messages not delivering

### ISSUE THREAD

**Issue by oxters168** [SteamRE/SteamKit#663]
Hey guys, so I'm having trouble with retrieving chat history and sending messages. Whenever I request chat history, I receive a fail result. When I try to send a message, nothing happens, but I do receive messages just fine though. I'm wondering if this is because there's a newer version of Steam possibly or if I'm doing something wrong.

**Comment by sp0ok3r** [comment]
I can send messages with no problems, cann we see some code?

**Comment by oxters168** [comment]
Definitely, I should've posted my code in the first message, sorry about that. [Here](https://pastebin.com/VEdAuNix)'s the function I use to initialize Steam. My [chat functions](https://pastebin.com/0cDGHp2C) are practically one line. [This](https://pastebin.com/yXwWba4J) is the class these functions are in, just in case needed. Then from a list item I've created for each friend, I have a button that calls [this function](https://pastebin.com/EPdPBGVE), which calls the chat functions I showed before. Also [here](https://pastebin.com/pdfkQuPe)'s the class for the list item. Here's an image to see what's happening in my application:
![ChatFails](https://user-images.githubusercontent.com/15348986/56131589-8a326a80-5f90-11e9-9e19-d0d0f9204a19.png)

The bottom right corner has the list items of friends each with it's own button. When the button is pressed, it calls the InitiateChat() function. In the green text above I show what is received after sending the request, and it shows the result is fail.

**Comment by sp0ok3r** [comment]
Try to see steamkit2 debug using:

```
 DebugLog.AddListener(new MyListener());
 DebugLog.Enabled = false;
```
```
class MyListener : IDebugListener
{
    public void WriteLine(string category, string msg)
    {
        Console.WriteLine("[IDebug] - " + category + ": " + msg);
    }
}
```
Example:
[DebugLogSample](https://github.com/SteamRE/SteamKit/blob/master/Samples/3.DebugLog/Program.cs#L30)

**Comment by sp0ok3r** [comment]
Other thing, can you try to send a message with the steamid defined: steamFriends.SendChatMessage(76561198041931474, EChatEntryType.ChatMsg, "hello princess");

**Comment by oxters168** [comment]
I'm working with Unity, so I don't receive console.writeline messages. Instead I used Unity's Debug.Log, which is practically the same thing. Anyways, on to the results...

The debug log messages I got after asking for message history:
[IDebug] - CMClient: Sent -> EMsg: ClientFSGetFriendMessageHistory (Proto: True)
[IDebug] - CMClient: <- Recv'd EMsg: Multi (1) (Proto: True)
[IDebug] - CMClient: <- Recv'd EMsg: ClientFSGetFriendMessageHistoryResponse (7526) (Proto: True)

The debug log messages I got after sending "hello princess" to the id you gave me:
[IDebug] - CMClient: Sent -> EMsg: ClientFriendMsg (Proto: True)
[IDebug] - CMClient: Sent -> EMsg: ClientHeartBeat (Proto: True)
[IDebug] - CMClient: <- Recv'd EMsg: Multi (1) (Proto: True)
[IDebug] - CMClient: <- Recv'd EMsg: ClientServiceCall (831) (Proto: True)
[IDebug] - CMClient: <- Recv'd EMsg: Multi (1) (Proto: True)
[IDebug] - CMClient: <- Recv'd EMsg: ClientClanState (822) (Proto: True)

I really appreciate you taking the time to try and help me, even if it doesn't lead to a solution, so thank you.

**Comment by oxters168** [comment]
I had been checking some messages before with the net hook analyzer and realized a different, longer steam id like the one you gave me before. I tested using that longer id of my own account, and it sent the message and received history! Thank you so much sp0ok3r! Also I found how to get that id from SteamKit. In case anyone later runs into a similar problem, the SteamID class has a function called ConvertToUInt64, that will give you the longer id version.
